server/app/lib/actions/fusor/deployment/deploy_rhev.rb

```diff
@@ -0,0 +1,226 @@
+#
+# Copyright 2015 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public
+# License as published by the Free Software Foundation; either version
+# 2 of the License (GPLv2) or (at your option) any later version.
+# There is NO WARRANTY for this software, express or implied,
+# including the implied warranties of MERCHANTABILITY,
+# NON-INFRINGEMENT, or FITNESS FOR A PARTICULAR PURPOSE. You should
+# have received a copy of GPLv2 along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+
+module Actions
+  module Fusor
+    module Deployment
+      class DeployRhev < Actions::Base
+        def humanized_name
+          _("Deploy Red Hat Enterprise Virtualization")
+        end
+
+        def plan(deployment)
+          Rails.logger.warn "XXX ================ Planning RHEV Deployment ===================="
+          Rails.logger.warn "XXX deploy_rhev? #{deployment.deploy_rhev}"
+          Rails.logger.warn "XXX deployment.id? #{deployment.id}"
+
+          # VERIFY PARAMS HERE
+          plan_self deployment_id: deployment.id
+
+          Rails.logger.warn "XXX plan_self called, leaving plan method"
+        end
+
+        def finalize
+          Rails.logger.warn "XXX ================ Deploy RHEV finalize method ===================="
+
+          deployment_id = input.fetch(:deployment_id)
+
+          Rails.logger.warn "XXX finalize: working with deployment: #{deployment_id}"
+
+          deployment = ::Fusor::Deployment.find(deployment_id)
+
+          Rails.logger.warn "XXX finalize: deployment is a: #{deployment.class}"
+
+          if deployment.rhev_engine_host.nil?
+            Rails.logger.warn "XXX finalize: WE DO NOT HAVE A RHEV ENGINE HOST."
+          else
+            Rails.logger.warn "XXX finalize: managed host? #{deployment.rhev_engine_host.managed?}"
+          end
+          Rails.logger.warn "XXX finalize: deployment name is: #{deployment.name}"
+          Rails.logger.warn "XXX finalize: deployment org is: #{deployment.organization.id}"
+
+          Rails.logger.warn "XXX ================ RHEV Hypervisor piece ===================="
+
+          # let's find the proper hostgroup and associate it with the
+          # hypervisors
+          hypervisor_group = find_hypervisor_hostgroup(deployment)
+
+          Rails.logger.warn "XXX finalize: calling assign_host_to_hostgroup (hypervisor)"
+
+          deployment.discovered_hosts.each do |da_host|
```

Comment:
```deployment.discovered_hosts.each``` worked without a hypervisor set. That's how I did most of my testing. It would be an empty list not a nil.